configure.ac

```
@@ -73,14 +73,14 @@ AC_LANG([C++])
 # But building inside a virtual machine environment has been found to
 # cause random Illegal Instruction errors due to mis-detection of CPU.
 AC_ARG_ENABLE(arch-native,
-  AS_HELP_STRING([--disable-arch-native],[Some compilers offer CPU-specific
+  AS_HELP_STRING([--enable-arch-native],[Some compilers offer CPU-specific
                  optimizations with the -march=native parameter.
-                 This flag disables the optimization. The default is to
-                 auto-detect compiler support and use where available.]), [
+                 This flag enables the optimization, by default disabled,
+                 provided that the compiler supports it.]), [
   SQUID_YESNO([$enableval],[--enable-arch-native])
 ])
-AC_MSG_NOTICE([CPU arch native optimization enabled: ${enable_arch_native:=auto}])
-AS_IF([test "x${enable_arch_native}" != "xno"],[
+AC_MSG_NOTICE([CPU arch native optimization enabled: ${enable_arch_native:=no}])
+AS_IF([test "x${enable_arch_native}" = "xyes"],[
   SQUID_CC_CHECK_ARGUMENT([squid_cv_check_marchnative],[-march=native])
```

Please either adjust this code to exit ./configure with an error if SQUID_CC_CHECK_ARGUMENT() fails to detect `-march=native` support OR add the following comment:

```suggestion
  # XXX: Fail if the explicitly requested native optimization is not available.
  SQUID_CC_CHECK_ARGUMENT([squid_cv_check_marchnative],[-march=native])
```

I strongly recommend the former (i.e. addressing this XXX in this PR) because it is not difficult to do and it is the right thing to do. I do not insist on that implementation because it should have been done earlier, when `--enable-arch-native` was added. This PR has the right to declare fixing that old bug out of scope.